pygeoapi/provider/oracle.py

```diff
@@ -54,6 +54,102 @@ class DatabaseConnection:
+        if DatabaseConnection.pool is None:
+            # dsn = oracledb.makedsn(
+            #     conn_dict["host"],
+            #     conn_dict.get("port", 1521),
+            #     service_name=conn_dict["service_name"]
+            # )
+            try:
+                if conn_dict.get("init_oracle_client", False):
+                    oracledb.init_oracle_client()
+
+                # Connect with tnsnames.ora entry and Login with Oracle Wallet
+                if conn_dict.get("external_auth") == "wallet":
+                    LOGGER.debug(
+                        "Oracle connect with tnsnames.ora entry \
+                        and login with Oracle Wallet"
+                    )
+
+                    if "tns_name" not in conn_dict:
+                        raise ProviderConnectionError(
+                            "tns_name must be set for external authentication!"
+                        )
+
+                    dsn = conn_dict["tns_name"]
+
+                # Connect with SERVICE_NAME
+                if "service_name" in conn_dict:
+                    LOGGER.debug(
+                        f"Oracle connect with service_name: \
+                            {conn_dict['service_name']}"
+                    )
+
+                    if "host" not in conn_dict:
+                        raise ProviderConnectionError(
+                            "Host must be set for connection with service_name!"
+                        )
+
+                    dsn = oracledb.makedsn(
+                        conn_dict["host"],
+                        conn_dict.get("port", 1521),
+                        service_name=conn_dict["service_name"],
+                    )
+
+                # Connect with SID
+                elif "sid" in conn_dict:
+                    LOGGER.debug(
+                        f"Oracle connect with sid: {conn_dict['sid']}"
+                    )
+
+                    if "host" not in conn_dict:
+                        raise ProviderConnectionError(
+                            "Host must be set for connection with sid!"
+                        )
+
+                    dsn = oracledb.makedsn(
+                        conn_dict["host"],
+                        conn_dict.get("port", 1521),
+                        sid=conn_dict["sid"],
+                    )
+
+                # Connect with tnsnames.ora entry
+                elif "tns_name" in conn_dict:
+                    LOGGER.debug(
+                        f"Oracle connect with tns_name: \
+                            {conn_dict['tns_name']}"
+                    )
+                    dsn = conn_dict["tns_name"]
+
+                else:
+                    raise ProviderConnectionError(
+                        "One of service_name, sid or tns_name must be specified!"
+                    )
+
+                LOGGER.debug(f"Oracle DSN string: {dsn}")
+
+                DatabaseConnection.pool = oracledb.create_pool(
```

Comment:
in the current design, it's not obvious  whose responsibility the pool is. it might be clearer if this function only returns the pool or if it check whether the pool is initialized as well as taking care of thing an initialization